htdocs/class/userutility.php

```diff
@@ -184,7 +184,9 @@ public static function validate()
         $uid    = is_object($user) ? $user->getVar('uid') : 0;
         $sql    = 'SELECT COUNT(*) FROM `' . $xoopsDB->prefix('users') . '` WHERE `uname` = ' . $xoopsDB->quote(addslashes($uname)) . (($uid > 0) ? " AND `uid` <> {$uid}" : '');
         $result = $xoopsDB->query($sql);
-        [$count] = $xoopsDB->fetchRow($result);
+        if ($result instanceof \mysqli_result) {
+            [$count] = $xoopsDB->fetchRow($result);
```

Comment:
[] in list unpacking is only allowed since PHP 7.1 

Please revert this change